be/src/exprs/agg/group_concat.h

```diff
@@ -611,106 +613,41 @@ class GroupConcatAggregateFunctionV2
-            Permutation perm;
-            Columns order_by_columns;
-            SortDescs sort_desc(ctx->get_is_asc_order(), ctx->get_nulls_first());
-            order_by_columns.assign(state_impl.data_columns->begin() + output_col_num, state_impl.data_columns->end());
-            Status st = sort_and_tie_columns(ctx->state()->cancelled_ref(), order_by_columns, sort_desc, &perm);
-            // release order-by columns early
-            order_by_columns.clear();
-            state_impl.release_order_by_columns();
-            if (UNLIKELY(ctx->state()->cancelled_ref())) {
-                ctx->set_error("group_concat detects cancelled.", false);
-                return;
-            }
-            if (UNLIKELY(!st.ok())) {
-                ctx->set_error(st.to_string().c_str(), false);
-                return;
-            }
-            for (auto i = 0; i < output_col_num; ++i) {
-                materialize_column_by_permutation(outputs[i].get(), {(*state_impl.data_columns)[i]}, perm);
-            }
-        }
-        // further remove duplicated values, pick the last unique one to identify the last sep and don't output it.
-        // TODO(fzh) optimize it later
-        std::vector<bool> duplicated(outputs[0]->size(), false);
-        if (ctx->get_is_distinct()) {
-            for (auto row_id = 0; row_id < elem_size; row_id++) {
-                bool is_duplicated = false;
-                for (auto next_id = row_id + 1; next_id < elem_size; next_id++) {
-                    bool tmp_duplicated = true;
-                    for (auto col_id = 0; col_id < output_col_num - 1; col_id++) { // exclude sep
-                        if (!outputs[col_id]->equals(next_id, *outputs[col_id], row_id)) {
-                            tmp_duplicated = false;
-                            break;
-                        }
-                    }
-                    if (tmp_duplicated) {
-                        is_duplicated = true;
-                        break;
-                    }
-                }
-                duplicated[row_id] = is_duplicated;
-            }
-        }
+
         // copy col_0, col_1 ... col_n row by row
         auto* string = down_cast<BinaryColumn*>(ColumnHelper::get_data_column(to));
         if (to->is_nullable()) {
             down_cast<NullableColumn*>(to)->null_column_data().emplace_back(0);
         }
         Bytes& bytes = string->get_bytes();
         size_t offset = bytes.size();
+        size_t old_offset = offset;
         size_t length = 0;
         std::vector<BinaryColumn*> binary_cols(output_col_num);
         for (auto i = 0; i < output_col_num; ++i) {
             auto tmp = ColumnHelper::get_data_column(outputs[i].get());
             binary_cols[i] = down_cast<BinaryColumn*>(tmp);
-            length += binary_cols[i]->get_bytes().size();
+        }
+        for (auto j = *state_impl.output_index;
+             j < elem_size && j < ((*state_impl.output_index) + ctx->state()->chunk_size()); ++j) {
```

Comment:
Why do we need to check chunk_size()? Does it mean that finalize_to_column() can be called multiple times with partial views?
Our implementation requires seeing all aggregated data in finalize_to_column() at once.

Reply:
I think what we do here is to support output result chunk by chunk.